packages/jasmine-runner/src/jasmine-runner.ts

```diff
@@ -97,43 +83,89 @@ class JasmineRunner implements TestRunner {
+                return new ExecutionResult();
             }
 
             if (!testFilesToProcessList) {
-                return new ExecutionResult(taskID, buildID, repoID, commitID, orgID);
+                return new ExecutionResult();
             }
+           
+            const jasmineObj = await this.createJasmineRunner(config);
 
-            const jasmineObj = await this.createJasmineRunner(argv.config);
             await this.loadSpecs(jasmineObj, testFilesToProcessList, entityIdFilenameMap);
+            const rootSuite = jasmineObj.env.topSuite();
             const specIdsToRun: number[] = [];
-            this.fetchSpecIdsToRun(jasmine.getEnv().topSuite(), specIdsToRun, entityIdFilenameMap,
+          
+            this.fetchSpecIdsToRun(rootSuite, specIdsToRun, entityIdFilenameMap,
                 testLocators, blockListedLocators);
-            if (specIdsToRun.length == 0) {
+            
+                if (specIdsToRun.length == 0) {
                 // pushing an invalid specID because if we pass empty array, it runs all specs
                 specIdsToRun.push(-1);
             }
             const reporter = new CustomReporter(runTask, entityIdFilenameMap);
-            jasmineObj.env.addReporter(reporter);
+            jasmine.getEnv().addReporter(reporter);
             await jasmine.getEnv().execute(specIdsToRun as unknown as jasmine.Suite[]);
-            const executionResults = await runTask.promise;
-            Util.handleDuplicateTests(executionResults.testResults);
+            const executionResult = await runTask.promise;
+            Util.handleDuplicateTests(executionResult.testResults);
             if (locators.length > 0) {
-                executionResults.testResults = Util.filterTestResultsByTestLocator(executionResults.testResults,
+                executionResult.testResults = Util.filterTestResultsByTestLocator(executionResult.testResults,
                     testLocators, blockListedLocators)
-                if (executionResults.testSuiteResults.length > 0) {
-                    executionResults.testSuiteResults = Util.filterTestSuiteResults(executionResults.testResults,
-                        executionResults.testSuiteResults)
+                if (executionResult.testSuiteResults.length > 0) {
+                    executionResult.testSuiteResults = Util.filterTestSuiteResults(executionResult.testResults,
+                        executionResult.testSuiteResults)
                 }
             }
-            if (postTestResultsEndpoint) {
-                await Util.makeApiRequestPost(postTestResultsEndpoint, executionResults);
+
+            
+            for (const spec of testFilesToProcessList) 
+            {
+                //if (path.includes(spec) )
+                {
+                    console.log("Clearing spec from require cache: " + spec);
+                    delete require.cache[spec];
+                }
```

Comment:
what is this?

Reply:
To fix jasmine re-load issue. in case of create new jasmine object

Reply:
I was talking about the code written. `if` was commented out and the body was written inside a block. Not a good piece of code.